engine/src/main/java/org/pentaho/di/trans/TransExecutionConfiguration.java

```diff
@@ -0,0 +1,857 @@
+      String value = XMLHandler.getTagValue( argNode, "value" );
+      if ( !Utils.isEmpty( name ) && !Utils.isEmpty( value ) ) {
+        arguments.put( name, value );
+      }
+    }
+
+    // Read the parameters...
+    //
+    Node parmsNode = XMLHandler.getSubNode( trecNode, "parameters" );
+    int nrParams = XMLHandler.countNodes( parmsNode, "parameter" );
+    for ( int i = 0; i < nrParams; i++ ) {
+      Node parmNode = XMLHandler.getSubNodeByNr( parmsNode, "parameter", i );
+      String name = XMLHandler.getTagValue( parmNode, "name" );
+      String value = XMLHandler.getTagValue( parmNode, "value" );
+      if ( !Utils.isEmpty( name ) ) {
+        params.put( name, value );
+      }
+    }
+
+    // IMPORTANT: remote preview and remote debugging is NOT yet supported.
+    //
+
+    replayDate = XMLHandler.stringToDate( XMLHandler.getTagValue( trecNode, "replay_date" ) );
+    safeModeEnabled = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "safe_mode" ) );
+    logLevel = LogLevel.getLogLevelForCode( XMLHandler.getTagValue( trecNode, "log_level" ) );
+    setLogfile = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "log_file" ) );
+    logFileName = XMLHandler.getTagValue( trecNode, "log_filename" );
+    setAppendLogfile = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "log_file_append" ) );
+    createParentFolder = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "create_parent_folder" ) );
+    clearingLog = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "clear_log" ) );
+    gatheringMetrics = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "gather_metrics" ) );
+    showingSubComponents = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "show_subcomponents" ) );
+    String sPassedBatchId = XMLHandler.getTagValue( trecNode, "passedBatchId" );
+    if ( !StringUtils.isEmpty( sPassedBatchId ) ) {
+      passedBatchId = Long.parseLong( sPassedBatchId );
+    }
+    runConfiguration = XMLHandler.getTagValue( trecNode, "run_configuration" );
+
+    Node resultNode = XMLHandler.getSubNode( trecNode, Result.XML_TAG );
+    if ( resultNode != null ) {
+      try {
+        previousResult = new Result( resultNode );
+      } catch ( KettleException e ) {
+        throw new KettleException( "Unable to hydrate previous result", e );
+      }
+    }
+
+    // Try to get a handle to the repository from here...
+    //
+    Node repNode = XMLHandler.getSubNode( trecNode, "repository" );
+    if ( repNode != null ) {
+      String repositoryName = XMLHandler.getTagValue( repNode, "name" );
+      String username = XMLHandler.getTagValue( repNode, "login" );
+      String password = Encr.decryptPassword( XMLHandler.getTagValue( repNode, "password" ) );
+
+      RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+      repositoriesMeta.getLog().setLogLevel( log.getLogLevel() );
+      try {
+        repositoriesMeta.readData();
+      } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)